### PR TITLE
Workflow scripts improvements

### PR DESCRIPTION
**Pull request by matteodelabre**
* Send Discord notifications when new commits are added to the stable and testing branches and result in a successful CI build.
* Factor out the logic for synchronizing with the remote server to a `sync-repository` action.
* Factor out the logic for fetching build dependencies to a `setup` action.

For repository maintainers: I set up a repo at <https://github.com/toltec-dev/workflow-tests> for testing the updated `stable.yml` and `testing.yml` workflows. They deploy to <https://toltec-dev.org/workflow-tests-stable> and <https://toltec-dev.org/workflow-tests-testing>.

### CODE REVIEW

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -10,19 +10,24 @@ jobs:
...
+                remote-path: ${{ secrets.REMOTE }}:/srv/toltec/testing
+            - name: Send notification to Discord
+              uses: ./.github/actions/discord-send
+              with:
+                channel: ${{ secrets.DISCORD_TESTING_CHANNEL_ID }}
+                token: ${{ secrets.DISCORD_TESTING_CHANNEL_TOKEN }}
+                title: New Toltec testing update available
+                link: https://toltec-dev.org/testing
+                color: 0xe3b341
+                message: ${{ github.event.head_commit.message }}
```
Comment (raisjn) [inline .github/workflows/testing.yml]: what happens if the discord secrets are unset? (i have a repo which builds and pushes to my own URL and am wondering if the build will fail if the secrets are missing)

**Review by raisjn — DISMISSED** [review]
looks good, nice refactoring

File: .github/workflows/testing.yml (hunk 1 above)
Comment (matteodelabre) [reply .github/workflows/testing.yml]: It does fail the build — see <https://github.com/toltec-dev/workflow-tests/runs/1708568012>. I’ll think about a workaround.

Comment (matteodelabre) [reply .github/workflows/testing.yml]: Okay, adding the `continue-on-error` flag makes it ignore errors from the Discord step. (See <https://github.com/toltec-dev/workflow-tests/runs/1708690940>.)

**Review by raisjn — APPROVED** [review]

**Pull request merged**